### PR TITLE
refactor(padding and margin utiliies): replace suffixes to use new sp…

### PR DESCRIPTION
**Pull request by markup-mitchell**
…acer- schema

All p and m utilities, t,b,r,l,x,y,s,e and responsive variants updated to use new spacer- schema

### CODE REVIEW

**Review by GemmaLouise — APPROVED** [review]

**Comment by markup-mitchell** [comment]
@duncansanderson @mjpvs can I get your eyes on this early doors if possible?  I want to get it into v4 before it needs further reconciliation. It's tedious, but we need to get it out the way.

**Review by mjpvs — APPROVED** [review]

File: src/components/caption/Caption.vue [hunk 1]
```diff
@@ -215,7 +215,7 @@ export default {
         }
 
         .vs-caption__caption-info{
-            padding: $spacer-100;
+            padding: $spacer-100 0;
```
Comment (duncansanderson) [inline src/components/caption/Caption.vue:218]: Should this be `$spacer-0` instead of `0`?

**Review by duncansanderson — CHANGES_REQUESTED** [review]

File: src/components/caption/Caption.vue (hunk 1 above)
Comment (duncansanderson) [reply src/components/caption/Caption.vue:218]: Just for consistency really, I noticed a PR the other day where 0s were changed to $spacer-0 but it's not a massive issue

**Comment by markup-mitchell** [comment]
I resisted the urge to anticipate "for consistency" as a reason, but I knew it was coming! IMO consistency is a means to an end and and we should be able to articulate what the trade-off is. I don't think we currently do use `$spacer-0` in a consistent way: a quick search for `margin:` returns literally dozens of examples of `0` being used - more, in fact, than with the variable, from what I can see. 

Anyway, I hope you don't mind that I went ahead and merged on the basis of the approvals I had just to get moving. I've created a ticket (DS-537) to replace `0` with the var as a separate task.

**Pull request merged**